src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -0,0 +1,141 @@
+                        }
+                    }
+                }
+            }
+        }
+
+        private static void ApplyAcronymsDictionary(MgmtConfiguration config)
+        {
+            foreach ((var key, var value) in config.AcroynmsDictionary)
+            {
+                CommonAcronyms.Add(key, value);
+            }
+        }
+
+        public static void UpdateAcronyms(this CodeModel codeModel, MgmtConfiguration config)
+        {
+            ApplyAcronymsDictionary(config);
+            foreach (var schema in codeModel.GetAllSchemas())
+            {
+                TransformSchema(schema);
+            }
+        }
+
+        public static IEnumerable<Schema> GetAllSchemas(this CodeModel codeModel)
+        {
+            return codeModel.Schemas.Choices.Cast<Schema>()
+                .Concat(codeModel.Schemas.SealedChoices)
+                .Concat(codeModel.Schemas.Objects)
+                .Concat(codeModel.Schemas.Groups);
+        }
+
+        private static void TransformSchema(Schema schema)
+        {
+            switch (schema)
+            {
+                case ChoiceSchema:
+                case SealedChoiceSchema:
+                    TransformBasicSchema(schema);
+                    break;
+                case ObjectSchema objSchema: // GroupSchema inherits ObjectSchema, therefore we do not need to handle that
+                    TransformObjectSchema(objSchema);
+                    break;
+                default:
+                    throw new InvalidOperationException($"Unknown schema type {schema.GetType()}");
+            }
+        }
+
+        private static void TransformBasicSchema(Schema schema)
+        {
+            TransformLanguage(schema.Language);
+        }
+
+        private static void TransformLanguage(Languages languages)
+        {
+            var originalName = languages.Default.Name;
+            var newName = originalName.EnsureNameCase();
+            languages.Default.Name = newName;
+        }
+
+        private static void TransformObjectSchema(ObjectSchema objSchema)
+        {
+            // transform the name of this schema
+            TransformBasicSchema(objSchema);
+            foreach (var property in objSchema.Properties)
+            {
+                TransformLanguage(property.Language);
+            }
+        }
+
+        private static IDictionary<string, string> CommonAcronyms = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+            { "os", "OS" },
+            { "ip", "IP" },
+            { "vm", "Vm" },
+            { "cpu", "Cpu" },
+            { "dns", "Dns" },
+            { "http", "Http" },
+            { "https", "Https" },
+        };
+
+        internal static string EnsureNameCase(this string name, bool lowerFirst = false)
+        {
+            var words = name.SplitByCamelCase().ToArray();
+            var builder = new StringBuilder();
+            for (int i = 0; i < words.Length; i++)
```

Comment:
Lets see if we can improve this a bit from perf perspective.  First lets measure how long this would take against an RP like network, if its more than ~5 seconds lets investigate ways to improve